### PR TITLE
[DHCP] Add try/exception when using psutil.process_iter

### PR DESCRIPTION
**Pull request by yaqiangz**
#### Why I did it
After got running process list by psutil, if some of the processes exited, but invoke `name()` / `cmdline()` function of them, it will raise NoSuchProcess exception
Fix issue https://github.com/sonic-net/sonic-buildimage/issues/19507

##### Work item tracking
- Microsoft ADO **(number only)**: 28682744

#### How I did it
1. Add try/exception in process execution
2. Change func `get_target_process_cmds` to `get_target_process` for reuse

#### How to verify it
UT passed

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-dhcp-utilities/dhcp_utilities/common/utils.py [hunk 1]
```diff
@@ -150,18 +150,21 @@ def _parse_table_to_dict(table):
     return ret
 
 
-def get_target_process_cmds(process_name):
+def get_target_process(process_name):
```
Comment (yaqiangz) [inline src/sonic-dhcp-utilities/dhcp_utilities/common/utils.py:153]: Modify this func for reuse

File: src/sonic-dhcp-utilities/dhcp_utilities/common/utils.py [hunk 2]
```diff
@@ -150,21 +150,38 @@ def _parse_table_to_dict(table):
...
-            res.append(proc.cmdline())
+        try:
+            if proc.name() == process_name:
+                res.append(proc)
+        except psutil.NoSuchProcess:
+            continue
     return res
 
 
+def get_process_cmd(proc):
```
Comment (Blueve) [inline src/sonic-dhcp-utilities/dhcp_utilities/common/utils.py:171]:
```
def get_target_process_cmds(process_name):
def get_target_process(process_name):
```
I suggest keep existing function without signature change.

**Review by Blueve — APPROVED** [review]

**Comment by yaqiangz** [comment]
Hi @yxieca could you please help to merge this PR?

**Comment by yxieca** [comment]
> Hi @yxieca @bingwang-ms could you please approve the backport request?

@yaqiangz can you confirm that this change has been tested on 202311 image?

**Pull request merged**